soda_machine.rb

```diff
@@ -7,12 +7,25 @@ def initialize(args = {})
   end
 
   def current_inventory_count
+    return @sodas.length
   end
 
   def find_soda(soda_brand)
+    soda = soda_brand.downcase.split(' ').join('_')
+    sodas.find {|x| x.brand == soda}
   end
 
   def sell(soda_brand)
+    changed_name = soda_brand.downcase.split(' ').join('_')
+    length = sodas.length
+    sodas.each do |i|
+      if sodas.length < length
+        break
+      else
+        sodas.delete(i.)
```

Comment:
Running line 25 breaks because of the `i.` expression, which the ruby interpreter parses as a syntax error.